demos/src/baz.rs

```diff
@@ -1,6 +1,14 @@
 #![allow(dead_code)]
 
 use metered::{metered, ErrorCount, HitCount, InFlight, ResponseTime};
+use thiserror::Error;
+
+#[metered::metered_error_variants(registry = BazErrorMetrics)]
```

Comment:
It's not 100% clear to me why we need to name the registry here.  I'd think the macro would generate a custom `ErrorCount` metric, for this error, that can be applied anywhere later.

That could also be a derive `ErrorCount` though derives are arguably a better fit for traits - so I'm fine with an attribute like:
```rust
#[metered::error_count(name = "BazErrorCount"]
```

I'm not sure if you can make it work with your design but that would be ideal not to refer to the registry here, there could be several registries using this type.

Reply:
Hey @magnet, I can see why this might be a bit consuming. I've renamed `metered_error_variants(registry...` to `error_count(name...`, cleaned up the example and added some more rustdoc that should clear some things up. Thanks for the review